windows/test-app.js

```diff
@@ -0,0 +1,258 @@
+//@ts-check
+const path = require("path");
+const fs = require("fs");
+
+const windowsDir = "windows";
+const reactNativeModulePath = findClosestPathTo(
+    path.join("node_modules", "react-native-windows")
+);
+const testAppNodeModulePath = findClosestPathTo(
+    path.join("node_modules", "react-native-test-app")
+);
+const srcRootPath = path.join(testAppNodeModulePath, windowsDir);
+const destPath = path.resolve("");
+
+function findClosestPathTo(fileOrDirName) {
+    let basePath = path.resolve("");
+    const rootDirectory = basePath.split(path.sep)[0] + path.sep;
+
+    while (basePath !== rootDirectory) {
+        const candidatePath = path.join(basePath, fileOrDirName);
+        if (fs.existsSync(candidatePath)) {
+            return path.relative("", candidatePath);
+        }
+        //Get parent folder
+        basePath = path.dirname(basePath);
+    }
+
+    return null;
+}
+
+function walk(current) {
+    if (!fs.lstatSync(current).isDirectory()) {
+        return [current];
+    }
+
+    const files = fs
+        .readdirSync(current)
+        .map((child) => walk(path.join(current, child)));
+    return [current, ...files.flat()];
+}
+
+/**
+ * Get a source file and replace parts of its contents.
+ * @param srcPath Path to the source file.
+ * @param replacements e.g. {'TextToBeReplaced': 'Replacement'}
+ * @return The contents of the file with the replacements applied.
+ */
+function resolveContents(srcPath, replacements) {
+    let content = fs.readFileSync(srcPath, "utf8");
+
+    return Object.keys(replacements).reduce((_, regex) => {
+        return _.replace(new RegExp(regex, "g"), replacements[regex]);
+    }, content);
+}
+
+//Binary files in React Native Test App Windows project
+const binaryExtensions = [".png", ".pfx"];
+
+/**
+ * Copy a file to given destination, replacing parts of its contents.
+ * @param srcPath Path to a file to be copied.
+ * @param destPath Destination path.
+ * @param replacements: e.g. {'TextToBeReplaced': 'Replacement'}
+ */
+function copyAndReplace(
+    srcPath,
+    destPath,
+    relativeDestPath,
+    replacements = {}
+) {
+    const fullDestPath = path.join(destPath, relativeDestPath);
+    if (fs.lstatSync(srcPath).isDirectory()) {
+        if (!fs.existsSync(fullDestPath)) {
+            fs.mkdirSync(fullDestPath);
+        }
+        return;
+    }
+
+    const extension = path.extname(srcPath);
+    if (binaryExtensions.indexOf(extension) !== -1) {
+        // Binary file
+        fs.copyFile(srcPath, fullDestPath, (err) => {
+            if (err) {
+                throw err;
+            }
+        });
+    } else {
+        // Text file
+        const srcPermissions = fs.statSync(srcPath).mode;
+        let content = resolveContents(srcPath, replacements);
```

Comment:
If we don't plan to replace `content`, we should make it `const`:
```suggestion
        const content = resolveContents(srcPath, replacements);
```